### ISSUE TITLE
Add domintro explanations for web developers

### ISSUE THREAD

**Issue by domenic** [whatwg/fetch#543]
I think the API would greatly benefit from some domintro boxes. This would allow you to add some developer-facing documentation for the more confusing concepts that end up manifesting in the API, like a response's type. The domintro could actually explain what each value you get back means. (I wasn't able to figure out the difference between "basic" and "default" myself.)

**Comment by annevk** [comment]
"default" is a response before any filtering has taken place. You can only get it with `new Response().type`.

If someone is interested in working on this that'd be great. I'm happy to help, but won't have bandwidth for the foreseeable future.

**Comment by domfarolino** [comment]
Hey, I'm interested in working on this with some guidance. I think it is a good idea and I could probably learn a lot helping out.

First naïve question: what is a domintro box?...I mean, it sounds like a semi-informal place to explain things in layman terms, but I just hadn't heard the term before.

**Comment by domenic** [comment]
Sorry about that!

When we say domintro box, we mean the green note boxes that give an informative for-web-developers description of the API. See https://dom.spec.whatwg.org/#interface-event below the IDL. They're called domintro boxes because they use `<dl class="domintro">`.

Note that I'm not sure we have super-great consensus on how to format the source for these things, because many of them were originally in a different spec pre-processor and the conversion to Bikeshed was kind of rocky for something that mixes together code, var, and Bikeshed cross-references. As such I think there are various inconsistencies. But that one I linked to, for the DOM Event classes, seems pretty solid, so patterning off of that makes sense.

**Comment by annevk** [comment]
I think even the Events section in DOM leaves a bit to be desired due to how Bikeshed works.

> `<dt><code><var>event</var> . {{Event/target}}</code>`

would be better as

> `<dt><code><var>event</var> . <a attribute for=Event>target</a></code>`

to avoid the nested `<code>`, but generally it indeed looks fine. (Also, not applicable to Events necessarily, if you need multiple paragraphs in any of the `<dd>` it's better they all have at least one `<p>` inside them.)

Another thing to note is that these do not need to be as detailed as the corresponding normative description. Many details can be left out. We just want to convey the high-level idea and the common usage pattern.

**Comment by domfarolino** [comment]
All domintro boxes:

 - [ ] Headers class
 - [ ] *Body mixin??* (domintro for this?)
 - [x] Request class (wip: https://github.com/whatwg/fetch/pull/617)
 - [ ] Response class.

**Comment by M-ZubairAhmed** [comment]
Hi @domfarolino are you working on adding domintro boxes to `Response`, i see its unchecked. If you are please go ahead. Let me know if you need a hand thank you

**Comment by annevk** [comment]
@M-ZubairAhmed feel free to take that on.

**Comment by domfarolino** [comment]
@M-ZubairAhmed Ah, unfortunately I haven't touched this in a while :( if you'd like to take on `Response` domintro box I'd be happy to review

**Comment by M-ZubairAhmed** [comment]
Thank you @annevk @domfarolino
